Mathlib/Order/ReltoGaloisConnection.lean

```diff
@@ -0,0 +1,107 @@
+/-
+Copyright (c) 2024 Lagrange Mathematics and Computing Research Center. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anthony Bordg
+-/
+import Mathlib.Data.Rel
+import Mathlib.Order.GaloisConnection
+
+/-!
+# The Galois Connection Induced by a Relation
+
+In this file, we show that an arbitrary relation `R` between a pair of types `α` and `β` defines
+a pair `l` and `u` of adjoint order-preserving maps between the corresponding posets
+`Set α` and `(Set β)ᵒᵈ`.
+We define `lFixedPoints` (resp. `uFixedPoints`) as the set of fixed points `J` (resp. `I`) of
+`Set α` (resp. `(Set β)ᵒᵈ`) such that `u (l J) = J` (resp. `l (u I) = I`).
+
+## Main Results
+
+⋆ `to_galoisConnection`: we prove that the maps `l` and `u` form a Galois connection.
+⋆ `instEquivFixedPoints`: we prove that the said maps induce inverse bijections between the sets
+of fixed points.
+
+## References
+
+⋆ Engendrement de topologies, démontrabilité et opérations sur les sous-topos, Olivia Caramello and
+Laurent Lafforgue (in preparation)
+
+## Tags
+
+relation, Galois connection, induced bijection, fixed points
+-/
+
+section
+
+universe u v
+
+variable {α : Type u} {β : Type v} (R : Rel α β)
+
+namespace Rel
+
+/-! ### Pairs of adjoint maps defined by relations -/
+
+open OrderDual
+
+/-- `l` maps any set `J` of elements of type `α` to the set `{b : β | ∀ a ∈ J, R a b}` of
+elements `b` of type `β` such that `R a b` for every element `a` of `J`. -/
+def l (J : Set α) : (Set β)ᵒᵈ := toDual {b : β | ∀ a ∈ J, R a b}
+
+/-- `u` maps any set `I` of elements of type `β` to the set `{a : α | ∀ b ∈ ofDual I, R a b}`
+of elements `a` of type `α` such that `R a b` for every element `b` of `I`. -/
+def u (I : (Set β)ᵒᵈ) : Set α := {a : α | ∀ b ∈ ofDual I, R a b}
+
+/-- The pair of functions `l` and `u` forms a Galois connection. -/
+theorem to_galoisConnection : GaloisConnection (l R) (u R) := by
```

Comment:
`to_galoisConnection` doesn't follow our naming conventions for mathlib.

Reply:
`OrderIso.to_galoisConnection` is actually misnamed as well. It should be `OrderIso.galoisConnection` or `OrderIso.gc`

Reply:
Ok, so it should simply be `Rel.galoisConnection` or without the prefix `galoisConnection`? Just for my personal enlightenment, which naming convention(s) `to_galoisConnection` doesn't follow?

Reply:
`to` is only used for forgetful inheritance of structures